test/test_types.py

```diff
@@ -1152,6 +1152,12 @@ def test_TPM2B_SIMPLE_OBJECT(self):
         b = bytes(dig)
         self.assertEqual(b, bob)
 
+        self.assertEqual(dig, bob)
+        self.assertNotEqual(dig, "pinchofbytes")
+
+        sdig = TPM2B_DIGEST(bob)
+        self.assertEqual(dig, sdig)
```

Comment:
what if you pass `TPM2B_ATTEST(bob)`, we want that to fail.

Reply:
why should it fail if it has the same value?

Reply:
My thought is that the type conveys information about the value, so the type should match. The API/Spec details different simple TPM2B's for context around the value, and that if the comparison involved two simple TPM2B's the instances should match around that context.

Reply:
My understanding/view is that the ```__eq__``` method is just for value comparison (so ==) while "is" is used for checking the type as well.
So for example:
```Python
>>> b = b"1"
>>> ba = bytearray(b"1")
>>> b == ba
True
>>> b is ba
False
>>> b2 = b"1"
>>> b is b2
True
>>> 
```

Reply:
> My understanding/view is that the `__eq__` method is just for value comparison (so ==) while "is" is used for checking the type as well.
> So for example:
> 
> ```python
> >>> b = b"1"
> >>> ba = bytearray(b"1")
> >>> b == ba
> True
> >>> b is ba
> False
> >>> b2 = b"1"
> >>> b is b2
> True
> >>> 
> ```

Ahh ok, I forgot that `is` exists.